test/timeout.ts

```diff
@@ -293,7 +293,7 @@ test.serial('secureConnect timeout', withServerAndFakeTimers, async (t, _server,
 			retry: 0
 		}).on('request', (request: http.ClientRequest) => {
 			request.on('socket', () => {
-				clock.runAll();
+				clock?.runAll();
```

Comment:
Why this change?

Reply:
In the HTTPS Test Server the clock is optional.
I don't see any reason to create a clock for each test, especially if there are no tests that require a "local" clock (not installed by `FakeTimers.install()`).

Reply:
https://github.com/sindresorhus/got/pull/1461#issuecomment-691153647

Reply:
Then you need to use `clock!.runAll();` since we expect `clock` to be defined.